backend/test/modules/tournament-tree/core/application/EnqueueMatchWhenTournamentWasStarted.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { TournamentWasStarted } from '../../../../../src/modules/doubles-tournament/core/domain/event/TournamentWasStarted';
+import { testTournamentTreeModule } from './TestTournamentTreeModule';
+import { NumberIdGeneratorStub } from '../../../../test-support/shared/core/NumberIdGeneratorStub';
+import { CreateTournamentTree } from '../../../../../src/modules/tournament-tree/core/application/command/CreateTournamentTree';
+import { testDoublesTournamentsModule } from '../../../doubles-tournament/core/application/TestDoublesTournamentsModule';
+import { FromListIdGeneratorStub } from '../../../../test-support/shared/core/FromListIdGeneratorStub';
+import { EnqueueMatch } from '../../../../../src/modules/doubles-tournament/core/application/command/EnqueueMatch';
+import { CreateTournamentWithTeams } from '../../../../../src/modules/doubles-tournament/core/application/command/CreateTournamentWithTeams';
+import { CommandBus } from '../../../../../src/shared/core/application/command/CommandBus';
+import { InMemoryCommandBus } from '../../../../../src/shared/infrastructure/core/application/command/InMemoryCommandBus';
+import { StoreAndForwardDomainEventBus } from '../../../../../src/shared/infrastructure/core/application/event/StoreAndForwardDomainEventBus';
+import { InMemoryDomainEventBus } from '../../../../../src/shared/infrastructure/core/application/event/InMemoryDomainEventBus';
+
+describe('Automated match enqueueing', () => {
+  it('When tournament was started, then enqueue all ready to start matches', async () => {
+    //Given
+    const currentTime = new Date();
+    const entityIdGenFromList = FromListIdGeneratorStub(['team1', 'team2', 'team3', 'team4']);
+    const entityIdGen = NumberIdGeneratorStub(100, 'entityId');
+    const commandBus: CommandBus = new InMemoryCommandBus();
+    const eventBus: StoreAndForwardDomainEventBus = new StoreAndForwardDomainEventBus(new InMemoryDomainEventBus());
```

Comment:
Czemu ten eventBus jest tutaj przekazywany, skoro jest taki jak default?

Reply:
inaczej test nie przechodzi :P jak wymyśliliśmy z Pawłem, chyba dlatego, że wtedy każdy z test modułów tworzy osobną instancję tego event busa

Reply:
Aaa, tak to prawda - dobra rozkmina :) Tak jest. Fajnie byłoby jakiejś kompozycji użyć i robić taki moduł testowy z dwóch modułów ale z braku laku może być :)

Reply:
Chocaiz nie dalo sie jakos tego rozgraniczyc, np. sprwadzajac dla jednego modulu tylko czy publikuje np event / wykonuje command, a w drugim od razu wrzucajac z palca ten event/command? Tak robilismy chyba zazwyczaj :) ?

Reply:
chodzi Ci o dwa osobne test case'y - 1. sprawdzamy czy moduł wykonał komendę StartTournament i wypluł event TouranemtnWasStarted 2. publikujemy TournamentWasStarted i sprawdzamy czy została wykonana komenda EqnueueMatch?

Reply:
No cos w tym stylu. Moze by bylo łątwiej i mamy test per moduł. Działałoby to, czy ja czegos nie widze?
Ale moze teraz nie zmieniaj, bo warto zajac sie czyms innym :)